tc/lang/lexer.h

```diff
@@ -67,15 +65,21 @@ namespace lang {
   _(TK_TIMES_EQ_B, "times_eq_b", "*=!")          \
   _(TK_MIN_EQ_B, "min_eq_b", "min=!")            \
   _(TK_MAX_EQ_B, "max_eq_b", "max=!")            \
-  _(TK_INT8, "int8", "int8")                     \
-  _(TK_INT16, "int16", "int16")                  \
-  _(TK_INT32, "int32", "int32")                  \
-  _(TK_INT64, "int64", "int64")                  \
+                                                 \
+  _(TK_BOOL, "bool", "bool")                     \
   _(TK_UINT8, "uint8", "uint8")                  \
   _(TK_UINT16, "uint16", "uint16")               \
   _(TK_UINT32, "uint32", "uint32")               \
   _(TK_UINT64, "uint64", "uint64")               \
-  _(TK_BOOL, "bool", "bool")                     \
+  _(TK_INT8, "int8", "int8")                     \
+  _(TK_INT16, "int16", "int16")                  \
+  _(TK_INT32, "int32", "int32")                  \
+  _(TK_INT64, "int64", "int64")                  \
+  _(TK_FLOAT16, "float16", "float16")            \
```

Comment:
Do we want to have _both_ `float32` and `float`?  Do we want to differentiate?

Reply:
Given the semantics for the other types it is consistent.
Removing float will break codes.
I don't see a big issue in the duplication here.